### PR TITLE
prevent duplicate actions from popping up

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Prevent duplicate actions from popping up in the maintenance.

- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)

### Testing & Verification

This change is already covered by existing tests, such as *all cluster DDL tests*.

https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5167/

### CODE REVIEW

**Comment by jsteemann** [comment]
tests blue

File: arangod/Cluster/MaintenanceFeature.cpp [hunk 1]
```diff
@@ -649,20 +645,6 @@ arangodb::Result MaintenanceFeature::storeIndexError(
   return Result();
 }
 
-arangodb::Result MaintenanceFeature::indexErrors(
```
Comment (maierlars) [inline arangod/Cluster/MaintenanceFeature.cpp]: This was unused code? Isn't that a problem?

Comment (jsteemann) [reply arangod/Cluster/MaintenanceFeature.cpp]: Problem solved by removing the code :smile:
Not sure what it was originally intended for. However, index errors are still captured and processed when the `copyAllErrors` method is called. Probably someone had plans to access just index errors separately at some point.

File: arangod/Cluster/ActionBase.cpp [hunk 2]
```diff
@@ -121,12 +121,12 @@ void ActionBase::createPreAction(std::shared_ptr<ActionDescription> const& descr
 
 /// @brief Retrieve pointer to action that should run before this one
 std::shared_ptr<Action> ActionBase::getPreAction() {
-  return (_preAction != nullptr) ? _feature.findAction(_preAction) : nullptr;
+  return (_preAction != nullptr) ? _feature.findFirstNotDoneAction(_preAction) : nullptr;
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/Cluster/ActionBase.cpp]: Yes, before a `done` action _may_ have been returned.
The change is exactly to return a not-done action here, to see if anything is still pending for the same type of operation. Anything that is already done I think we can ignore.

At least that was what Max said we should do. Maybe I misunderstood something.

**Pull request merged**